docs: restructure Layered Zero Trust documentation (#720)
validatedpatterns/docs: content/patterns/layered-zero-trust/lzt-getting-started.adoc

.Prerequisites

* An {ocp} 4.20, 4.21, or 4.22 cluster with:

> .. publicly signed certificates for Ingress.

Reviewer: We do not require publicly signed certificates anymore and support both public/self signed certificates

.. default `StorageClass` which provides dynamic `PersistentVolume` storage.

* Fulfill the general link:https://validatedpatterns.io/learn/quickstart/#_prerequisites[prerequisites for Validated Patterns].

> * Depending on the characteristics of your cluster, you might need additional hardware resources for the {rh-rhacm-first} component.

Reviewer: ```suggestion
* Depending on the deployed pattern components, you might need additional hardware resources.
```

> For a single-node cluster, you can start with 4 vCPUs, 16 GB of memory, and 120 GB of storage.

Reviewer 2: Let's take this opportunity to update this and align it with VP's requirements

```suggestion
For a single-node cluster, you can start with 8 CPUs, 16 GB of memory, and 120 GB of storage.
```

+
For more details about {rh-rhacm} sizing, see link:https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.14/html-single/install/index#sizing-your-cluster[Sizing your cluster].

* For the exact component versions validated on each OCP release, see the link:https://github.com/validatedpatterns/layered-zero-trust/blob/main/docs/support-matrix.md[Support Matrix].

[NOTE]
====

> The Layered Zero Trust pattern's default deployment assumes that none of its components have been installed previously. Verify that your {ocp} environment does not already contain any of xref:/patterns/layered-zero-trust/index.html#architecture[the listed components] before proceeding.

Reviewer 2: This link (xref:/patterns/layered-zero-trust/index.html#architecture) is broken, it now has its own page (lzt-architecture.adoc)

====

[id="lzt-gui-install"]
== Quick install via OpenShift Console

The fastest way to deploy the pattern is through the OpenShift Console. This GUI-based deployment covers the "batteries included" baseline configuration. For advanced feature sets, customization, or using your own fork, see the <<lzt-cli-install,Advanced Installation (CLI)>> section below.

> video::layered-zero-trust/gui-install/vp-1-click-install.mkv[width=800,opts="autoplay,loop,nocontrols"]

Reviewer: We should not be including a video that is stored within a git repository

[id="lzt-gui-step1"]
=== Step 1: Install the Validated Patterns Operator

> Navigate to the Pattern Catalog in the OpenShift Console: *Patterns > Catalog* in the left navigation. Check the *Tested* checkbox to filter patterns by tier, then click *Install* on the Layered Zero Trust card.

Reviewer 2: The **patterns** entry is not present by default in OpenShift, it's a console plugin installed using the Validated Patterns Operator.

The steps are:
1. Install the Validated Patterns Operator (this will reload the console).
2. Once reloaded, you'll see the Patterns option in the left-hand menu.

image::layered-zero-trust/gui-install/Validated_Pattern_Operator_Installation-step1.png[Validated Patterns Operator in the Software Catalog]

[id="lzt-gui-step2"]
=== Step 2: Review operator details

The operator detail page shows the channel (fast), version (latest), capability level (Basic Install), and provider (Red Hat).

image::layered-zero-trust/gui-install/Validated_Pattern_Operator_Installation-step2.png[Operator details page]

[id="lzt-gui-step3"]
=== Step 3: Configure and install the operator

Accept the defaults: All namespaces, `patterns-operator` namespace (auto-created), Automatic updates. Click *Install*.

image::layered-zero-trust/gui-install/Validated_Pattern_Operator_Installation-step3.png[Operator installation configuration]

[id="lzt-gui-step4"]
=== Step 4: Operator installed

Wait for the "Operator installed successfully" confirmation.

image::layered-zero-trust/gui-install/Validated_Pattern_Operator_Installation-step4.png[Operator installed successfully]

[id="lzt-gui-step5"]
=== Step 5: Install the Layered Zero Trust pattern

> Navigate to the Validated Patterns Operator and click *Install Pattern*. Configure:

Reviewer 2: This is where we go to the "patterns" menu option

> Navigate to the Validated Patterns Operator and click *Install Pattern*. Configure:

Reviewer 2: By default, **layered-zero-trust** is not displayed in the catalog because only `maintained` items are shown. Those listed as `tested` must also be added to the top checkbox.

* *Target Revision*: `main` (or your branch)
* *Secrets Configuration*: Expand each section to configure secrets (config-demo, qtodo-db, qtodo-truststore, acs-central, keycloak, rhtpa-db, rhtpa-oidc-cli, keycloak-users, quay-users). Secrets marked "Auto-generate this value" will be created automatically by Vault.

image::layered-zero-trust/gui-install/Layred_Zero_Trust_Installation.png[Layered Zero Trust pattern installation form]

image::layered-zero-trust/gui-install/Layered_Zero_Trust_Installation-optional-settings.png[Optional settings and secrets configuration]

Click *Install* to deploy the pattern.

[id="lzt-gui-step6"]
=== Step 6: Monitor deployment in Argo CD

Open the Argo CD console to monitor the deployment. Navigate to the proper Argo CD instance using the *Application Selector* (the icon with nine small squares) in the top navigation bar.

The Layered Zero Trust pattern installs two Argo CD instances on your Hub cluster:

* *Cluster Argo CD*: Manages the `layered-zero-trust-hub` App-of-Apps application. This deploys the pattern's components in the correct order.
* *Hub Argo CD*: Manages the Cluster Argo CD instance itself and the individual component applications.

> Navigate to the *Cluster Argo CD* instance to monitor the deployment progress. Components deploy in a specific order (sync waves). Initial deployment takes approximately 30-45 minutes.

Reviewer 2: I would remove the time. It depends on the size of your cluster. In my case, it takes approximately 15-20 minutes. I would use a more fuzzy value.

> Navigate to the *Cluster Argo CD* instance to monitor the deployment progress. Components deploy in a specific order (sync waves). The deployment time varies depending on your cluster size and resources.

Reviewer: ```suggestion
Navigate to the *Hub Argo CD* instance to monitor the deployment progress. Components deploy in a specific order (sync waves). The deployment time varies depending on your cluster size and resources.
```

> Navigate to the *Cluster Argo CD* instance to monitor the deployment progress. Components deploy in a specific order (sync waves). The deployment time varies depending on your cluster size and resources.

Reviewer: we should also mention how to login (ie make sure to use the "Login with OpenShift" option"

image::layered-zero-trust/gui-install/ArgoCD_Installation_Monitoring.png[Argo CD deployment monitoring]

[id="lzt-gui-step7"]
=== Step 7: Verify all components are healthy

When all Argo CD applications show *Healthy* and *Synced* status, the pattern is fully deployed.

image::layered-zero-trust/gui-install/ArgoCD_Installation_Monitoring-All-green.png[All Argo CD applications healthy]

[id="lzt-cli-install"]
== Advanced installation (CLI)

For advanced feature sets, customization, or when using your own fork, use the CLI-based installation with the `pattern.sh` script. This method provides full control over the deployment configuration.

.Additional prerequisites

* A GitHub account and a token with repository permissions, to read from and write to your forks.
* Access to Podman (or Docker) for execution of the container images used by the `pattern.sh` script for provisioning.

> * (Optional) The Helm binary, for instructions, see link:https://helm.sh/docs/intro/install/[Installing Helm].

Reviewer 2: why do they need Helm? I use it for development and debugging, but does a user really need it for installation? If it's not required, I would remove this line

[id="lzt-verify-deployment"]
=== Verify the deployment

The Layered Zero Trust pattern provisions every component and manages them through {ocp} GitOps. After you deploy the pattern, verify that all components are running correctly.

The Layered Zero Trust pattern installs the following two Argo CD instances on your Hub cluster. You can view these instances in the {ocp} web console by using the *Application Selector* (the icon with nine small squares) in the top navigation bar.

. *Cluster Argo CD*: Deploys an *App-of-Apps* application named `layered-zero-trust-hub`. This application installs the pattern's components.
. *Hub Argo CD*: Manages the Cluster Argo CD instance and the individual components that belong to the pattern on the hub {ocp} instance.

If every Argo CD application reports a *Healthy* status, the pattern has been deployed successfully.

[id="lzt-importing-existing-clusters"]

> == Importing existing clusters

Reviewer: Since we really do not utilize imported clusters, should we still keep this content or drop it?
